library/core/benches/char/methods.rs

```diff
@@ -35,3 +35,13 @@ fn bench_to_digit_radix_var(b: &mut Bencher) {
             .min()
     })
 }
+
+#[bench]
+fn bench_ascii_to_uppercase(b: &mut Bencher) {
+    b.iter(|| (0..=255).cycle().take(10_000).map(|b| char::from(b).to_uppercase()).count())
```

Comment:
function name says "ascii" but ascii would be only 0..=127 - so this is a benchmark that mixes the ascii and non-ascii case.

Reply:
Would it be possible to say anything about benchmarks results on mixed and non-ascii text? I expect branch prediction would mean that on pure non-ascii, there is virtually no change(?)

Reply:
I would expect so. What do you think of adding two more benchmarks for a total of three?
* all ASCII characters
* no ASCII characters
* 50/50 mixed characters

Reply:
We need to be slightly careful here. The non-ascii benchmark is always one byte unicode is it not? It is non-ascii but above 192 it's going to be indicating another code point so there's going to be pleanty of malformed unicode in here.
It might be slightly better to create a random slice of u8 and then use String::from_utf8_lossy() to make it valid first?

Reply:
Since it's a benchmark I would want to make the "random chars" constant so runs can be compared.

What do you think of this? [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b73bef6088be9d94949b35b5612e4dc5) Each test case can use the same list of chars and filter for what it's testing for (where mixed does no filtering).

Reply:
Random was the wrong choice of words here, sorry! It looks like a fair few other benchmarks / tests don't worry about putting in malformed utf sequences so I think it's fine to define the benchmarks this way for now.